### DISCUSSION TITLE
Add auth listeners to SDK

### DISCUSSION THREAD

**Discussion by apstanisic** [directus/directus#3673] [Feature Requests]
Something like Firebase: `firebase.auth().onAuthStateChanged`. 
I've added a couple example bellow, and can create PR.
```ts
// Adding
sdk.auth.onLogin =  (data: AuthResponse) => {};
sdk.auth.onLogout = () => {};
sdk.auth.onRefresh = (data: AuthResponse) => {};
sdk.auth.onChange = (action: 'login' | 'logout' | 'refresh', data?: AuthResponse) => {};
```

**Comment by rijkvanzanten** [comment]
sounds good to me!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
